api2-model/src/main/java/eu/europeana/api2/model/utils/Api2UrlService.java

```diff
@@ -187,4 +189,18 @@ public String getRedirectUrl(String wskey, String isShownAtUri, String provider,
         return url.toString();
     }
 
+    /**
+     * Generates URL path to search result record.
+     * /api/v2/ path prefix not required if running on the live environment (EA-2151)
+     *
+     * TODO: Remove hardcoded url check
+     * @return string array with path to record.
+     */
+    private String[] getApiRecordPath() {
+        if (!API_BASEURL.equals(api2BaseUrl)) {
```

Comment:
This check will return false if api2BaseUrl is not defined. Instead reading api2BaseUrl you could use the `getApi2BaseUrl()` method.